src/runtime/extended-access/gaa-metering.ts

```diff
@@ -454,16 +474,26 @@ export class GaaMetering {
       noIssues = false;
     }
 
-    // Check userState is an 'object'
-    if (
+    // For client-side paywall, either userState or publisherEntitlementPromise needs to provided
+    // - If granted is not provided in a userState, publisherEntitlementPromise needs to be provided.
+    // For server-side paywall, userState needs to be provided.
+    if ('userState' in params && typeof params.userState !== 'object') {
+      debugLog(`userState is not an object`);
+      noIssues = false;
+    } else if (
+      params.paywallType == PaywallType.SERVER_SIDE ||
+      params.showcaseEntitlement
+    ) {
+      if (!('userState' in params)) {
+        debugLog('userState needs to be provided');
```

Comment:
Could you add a test covering this?

Reply:
Done - added test cases to cover this scenario.